### DISCUSSION TITLE
Constraint maya transform to usdPrim

### DISCUSSION THREAD

**Discussion by kreppene** [Autodesk/maya-usd#1652] [Q&A] [answered]
Hi, in this video it is mention that it is possible to constraint a maya object to a usdPrim, using a python script
https://youtu.be/z-w0BktAA-0?t=1497

Would it be possible to get a simply example on how this can be done?

Thanks :)

**Comment by santosd** [comment, marked as answer]
Hello @kreppene,

We have a regression test that covers some of these things, which may give some insight as to how it could be done:
https://github.com/Autodesk/maya-usd/blob/dev/test/lib/testMayaUsdProxyAccessor.py#L107

Some extras:
Way to expose a translate operation to allow connection through the node editor:

```
import mayaUsd.lib.proxyAccessor as pa

selUfeObject = pa.getUfeSelection()
pa.createXformOps(selUfeObject)
pa.getOrCreateAccessPlug(ufeObject=selUfeObject,usdAttrName='xformOp:translate')
```

Parenting a Maya object to a USD Prim (select DAG child object\s first and then select the USD prim):
```
import mayaUsd.lib.proxyAccessor as pa

pa.parent()
```

Make note that we haven't fleshed out any workflows at this time, we have simply exposed some things that allow for possible ways to connect between MayaUSD and Maya DAG objects.

I hope that helps.

-David Santos

**Reply by kreppene** [reply]
Thank you @santosd, it works perfectly:)

**Reply by neilrubie** [reply]
I'm also experimenting with parenting Maya objects to usdPrims. While the parent() function exposed in proxyAccessor.py does seem to achieve the desired effect in terms of final transformations, the maya dag node itself stays where it is in the outliner, not _visibly_ parented underneath the prim. I'm just wondering if is this expected behaviour at this point? or am I missing something? Thanks!

**Reply by kxl-adsk** [reply]
Yes, this is expected. With UFE v3 we have added a way to replace a subset of the hierarchy with a Maya's one...but it's not hooked to the parenting example you are using.

**Reply by neilrubie** [reply]
> Yes, this is expected. With UFE v3 we have added a way to replace a subset of the hierarchy with a Maya's one...but it's not hooked to the parenting example you are using.

Now that UFE v3 has been released through maya2023, would you be able to point me to where I can find this hierarchy replacement functionality and how to use it? Specifically I'm looking to "parent" a maya hierarchy underneath a prim, similar to how it's being done with the mayaUsd's MayaReference node.

**Reply by kxl-adsk** [reply]
Hi @neilrubie. I no longer work at Autodesk, so please wait for confirmation from the team.

What I was referring to was [PathMapping](https://github.com/Autodesk/maya-usd/blob/dev/lib/mayaUsd/ufe/UsdPathMappingHandler.cpp#L110) service we added to support [UpdaterManager](https://github.com/Autodesk/maya-usd/blob/dev/lib/mayaUsd/fileio/primUpdaterManager.h). EditAsMaya workflow is utilizing this codepath.

Cheers,
KxL
